internal/services/jobs/jobs.go

```diff
@@ -157,6 +214,33 @@ func ResourceJobDefinitionRead(ctx context.Context, d *schema.ResourceData, m in
 		return diag.FromErr(err)
 	}
 
+	rawSecretRefs, err := api.ListJobDefinitionSecrets(&jobs.ListJobDefinitionSecretsRequest{
+		Region:          region,
+		JobDefinitionID: id,
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	secretRefs := make([]interface{}, len(rawSecretRefs.Secrets))
+
+	for i, secret := range rawSecretRefs.Secrets {
+		secretRef := make(map[string]interface{})
+		secretRef["secret_id"] = secret.SecretManagerID
+		secretRef["secret_reference_id"] = secret.SecretID
+		secretRef["secret_version"] = secret.SecretManagerVersion
+
+		if secret.File != nil {
+			secretRef["file"] = secret.File.Path
+		}
+
+		if secret.EnvVar != nil {
+			secretRef["environment"] = secret.EnvVar.Name
+		}
+
+		secretRefs[i] = secretRef
+	}
+
```

Comment:
Maybe move this flattening logic inside a dedicated function in a separate types.go file